crates/holochain/src/core/workflow/sys_validation_workflow/validate_op_tests.rs

```diff
@@ -94,10 +97,20 @@ async fn validate_dna_op_mismatched_dna_hash() {
     let outcome = test_case.with_op(op).run().await.unwrap();
 
     assert!(
-        matches!(outcome, Outcome::Rejected),
+        matches!(outcome, Outcome::Rejected(_),),
         "Expected Rejected but actual outcome was {:?}",
         outcome
     );
+    match outcome {
+        Outcome::Rejected(reason) => {
+            assert!(
+                reason.contains("The DNA does not belong in this space!"),
+                "Reason message does not match [{}]",
+                reason
+            );
+        }
+        _ => unreachable!(),
+    }
```

Comment:
A string literal needs to be changed when the error message is changed.
```suggestion
    match outcome {
        Outcome::Rejected(reason) => {
            let expected_reason = &ValidationOutcome::WrongDna(
                dna_action.hash,
                test_case.dna_def_hash().as_hash().to_owned(),
            )
            .to_string();
            assert!(
                reason.contains(expected_reason),
                "Reason message does not match [{}]",
                reason
            );
        }
        _ => unreachable!(),
    }
```

Reply:
This is a good suggestion, I'll work through as many of these as make sense. Some of the types are hard to construct :)

Reply:
This was even better than I thought, I've been able to really clean up the error asserting without losing quality error messages :) Done through the whole file!